packages/frontend/src/creation-form/payload-form/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { type FC } from "react";
+import {
+    Metric,
+    type PayloadFormProps as InnerPayloadForm,
+    type Specification,
+} from "../../types";
+import { TvlPayloadForm } from "./tvl";
+import { Typography } from "@carrot-kpi/ui";
+
+export interface PayloadFormProps extends InnerPayloadForm {
+    metric?: Metric | null;
+    payload?: Partial<Specification["payload"]>;
+    onChange: (payload: Specification["payload"]) => void;
+}
+
+const PAYLOAD_FORM_BY_METRIC: {
+    [M in Metric]: FC<PayloadFormProps>;
+} = {
+    [Metric.TVL]: TvlPayloadForm,
+};
+
+export const PayloadForm = ({
+    metric,
+    payload,
+    onChange,
+    ...rest
+}: PayloadFormProps) => {
+    if (!metric)
+        return (
+            <div className="flex justify-center">
+                <Typography>Select a metric</Typography>;
```

Comment:
Left a `;` at the end of this line.

Reply:
Addressed in a27d86bea05fac63494443db2a86b72af5f16e6f